src/v/datalake/translation/translation_probe.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "metrics/metrics.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+
+namespace datalake {
+
+class translation_probe final {
+    struct invalid_record_label_set {
```

Comment:
nit: could use docs, but also per my other comment I think I need some convincing that we need this